src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,17 +2638,508 @@ def test_binaryoperation_can_be_printed():
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("array1").name == "array1"
+    assert fake_parent.symbol_table.lookup("array1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("array1").shape == [None, None,
+                                                               None]
+    assert fake_parent.symbol_table.lookup("array1").access == 'local'
+
+    reader = FortranStringReader("real, dimension(:), intent(in) :: array2")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("array2").name == "array2"
+    assert fake_parent.symbol_table.lookup("array2").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("array2").shape == [None]
+    assert fake_parent.symbol_table.lookup("array2").access == 'read_arg'
+
+    reader = FortranStringReader("real, intent(in), dimension(:) :: array3")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("array3").name == "array3"
+    assert fake_parent.symbol_table.lookup("array3").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("array3").shape == [None]
+    assert fake_parent.symbol_table.lookup("array3").access == 'read_arg'
+
+    reader = FortranStringReader("integer, dimension(3,5) :: array4")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("array4").name == "array4"
+    assert fake_parent.symbol_table.lookup("array4").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("array4").shape == [3, 5]
+    assert fake_parent.symbol_table.lookup("array4").access == 'local'
+
+    reader = FortranStringReader("integer, dimension(*) :: array5")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert "Assumed-size arrays are not supported." in str(error.value)
+
+    reader = FortranStringReader("integer, dimension(var1) :: array5")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert ("Only integer literals are supported for explicit shape array"
+            " declarations.") in str(error.value)
+
+
+def test_fparser2astprocessor_handling_assignment_stmt(f2008_parser):
     ''' Test that fparser2 Assignment_Stmt is converted to expected PSyIRe
```

Comment:
Although you've not strictly touched it in this Issue, please could you do s/PSyIRe/PSyIR/ in the docstrings of these tests.

Reply:
Done